daffodil-tdml-lib/src/main/scala/org/apache/daffodil/tdml/TDMLRunner.scala

```diff
@@ -874,31 +890,45 @@ case class ParserTestCase(ptc: NodeSeq, parentArg: DFDLTestSuite)
         actual.getBlobPaths.foreach { blobPath =>
           Files.delete(blobPath)
         }
+        actualSax.getBlobPaths.foreach { blobPath =>
+          Files.delete(blobPath)
+        }
 
         val isErr: Boolean =
-          if (actual.isProcessingError) true
+          if (actual.isProcessingError || actualSax.isProcessingError ) true
```

Comment:
We want isError to be true only if both actual and actualSax errored. If one errored but the other didn't that means something went wrong.

I'm thinking all the checking logic should be split off into separate functions, then the of of this can be something like
```scala
checkParseResult(actual)
checkParseResult(actualSax)
```
That minimize duplicate code and hard to catch logic issues like this.